bus/uploadingsectors_test.go

```diff
@@ -0,0 +1,70 @@
+package bus
+
+import (
+	"errors"
+	"testing"
+
+	"go.sia.tech/core/types"
+	"go.sia.tech/renterd/api"
+	"lukechampine.com/frand"
+)
+
+func TestUploadingSectorsCache(t *testing.T) {
```

Comment:
Can we add a test with an actual upload? What I had in mind is the following:

1. Upload a random chunk (exactly 40MiB)
2. Check that the right number of roots are registered as ongoing
3. Let the upload finish
4. Check that the same roots are now returned as part of the contracts

The way to achieve this easily (I hope) is to use a `bytes.Buffer` for the upload instead of a `bytes.NewReader`. It should block after reading the chunk from it which causes the chunk to be uploaded but the upload not actually finishing. Then you can close the buffer to finish the upload.

Reply:
Hm good idea, I actually thought about this but never considered blocking on the reader. Sometimes I miss our interrupt system, I wonder when we'll end up adding that.